api/agent.go

```diff
@@ -395,15 +396,30 @@ func NewControllerWrapper(config *loadtest.Config, controllerConfig interface{},
 			username = creds[id].username
 			email = creds[id].email
 			password = creds[id].password
+
+			// Check if the user has a custom authentication type. Custom authentication types are
+			// specified by prepending the username with the authentication type followed by a colon.
+			// Example: "openid:user1@test.mattermost.com user1password"
+			if strings.Contains(username, ":") {
+				usernameParts := strings.Split(username, ":")
+				authenticationType = usernameParts[0]
+				username = usernameParts[1]
+
+				// Fix the email as well
+				emailParts := strings.Split(email, ":")
+				email = emailParts[1]
```

Comment:
Why is this needed? I thought just the username would become like "openid:user1"? The email should be untouched right? Also, we need to do length validation of the slice here to prevent panics.

Reply:
Since the `users.txt` file works with `email password` lines, what this does is parse if theres a `:` in the email to check for the auth service, hence parsing: `authservice:email password`. But in order to properly set the email I need to discard the `authservice:` part.

Reply:
I thought the authservice is just a prefix for the username right? Or is it for both?

Reply:
It's used in the users txt file only, but ignored for login.

Unsure how the `:` would affect login in, and I selected that symbol so we easily know in code if there's a custom authservice setup. This could be changed to use `openid-username-00001` and check if the first part split by `-` matches an authservice but it seemed more complicated in my head. Would love your fresh eyes input on this.